blog/2023-12-22-openhab-4-1-release.md

```diff
@@ -0,0 +1,245 @@
+
+The add-on suggestions feature currently checks mDNS and UPnP traffic and performs a simple IP scan on your network to find out which devices you have at home.
+Additionally, the list of running processes of your openHAB server is checked to suggest for example persistence services like InfluxDB.
+
+Many thanks to Mark Herwege ([@mherwege](https://github.com/mherwege)), Andrew Fiddian-Green ([@andrewfg](https://github.com/andrewfg)) and Holger Friedrich ([@holgerfriedrich](https://github.com/holgerfriedrich)) for implementing that awesome new feature!
+
+## Core Runtime Enhancements
+
+### Extending the Ontology with your own Semantic Tags
+
+_Laurent Garnier, openHAB Maintainer_
+
+You can now easily extend the default ontology with your own semantic tags and then use them to build your semantic model.
+This can be used for example to add a location or equipment that does not (yet) exist by default.
+
+To define your own semantic tags, create a YAML configuration file containing the description of all your specific semantic tags in the `$OPENHAB_CONF/tags` folder.
+You can find a [tutorial on the community forum](https://community.openhab.org/t/oh-4-1-tutorial-to-manage-custom-semantic-tags/148135).
+
+### Voice System Enhancements
+
+_Miguel Álvarez Díez, openHAB Contributor_
+
+Some minor improvements were introduced in the built-in voice system components:
+
+The standard text interpreter now uses the Item's synonyms metadata as alternative names, which helps to avoid name collisions.
+It also automatically scans your Item's command description metadata, so that the rule `Set <ItemLabel> to <ItemCommandLabel>` is supported out-of-the-box for all supported languages.
+
+Additionally, the `voiceSystem` metadata namespace was created to allow adding custom rules, for example:
+
+- You can add the rule `go to mode $cmd$ on $name$` to an Item labeled `Vacuum` with command description `MODE_CLEAN=cleaning`:
+  That will instruct the interpreter to send the command `MODE_CLEAN` to the Item on the phrase `go to mode cleaning on vacuum`.
+- You can also capture free text with a rule like `watch $*$ on $name$`, so that the phrase `watch my favorite show on tv` will send the command `my favorite show` to the Item labeled as `TV`.
+
+Two new configuration parameters where added to the dialog processor start/register command:
+
+- The `dialogGroup` option prevents simultaneous processing of dialogs on the same group, so that several speakers located near-by won't be triggered at the same time.
+- The `locationItem` option allows you to forward the location of a speaker/dialog processor to the standard interpreter.
+  This makes the interpreter prioritize child Items of the location provided, so that you can use "Light" as Item label for two Items in different two different rooms with a speaker each, and the phrase `Turn on the light` will target the light Item of the correct room.
+
+## Main UI Enhancements
```

Comment:
```suggestion
### Units of Measurement: currencies have arrived

_Jan N. Klug, openHAB Maintainer_

openHAB had good support for handling units in the past and that made a big step forward in ease-of-use with openHAB 4.

What was still missing (especially for energy management) was support for currencies.
We have now added two new dimensions `Currency` and `EnergyPrice`.

Currencies are different from "usual" units.
There are a lot more (each country has a different one) and even worse: their "conversion factor" is not fixed.

Think of length measurement: basically there is the metric and the imperial system.
The conversion from `mm` to `inch` is `25.4 mm = 1 inch`.
This has been the same ten years ago and will also be the same in 2050.
The factor (exchange rate) changes for Euro to US-Dollar on a very short time-scale.

It's even more complicated than that: A lot of countries use similar symbols (e.g. `$`) for their currency, but they mean different things: a Canadian dollar is not the same as an US dollar.
Luckily the ISO comes to our help.
ISO 4217 defines three letter "currency codes", unique for major currency.

openHAB supports a new type of service: the `CurrencyProvider`.
A `CurrencyProvider` can support one or more currencies (with at least the "currency code" as identifier) and brings the information how to convert between them.

Out of the box openHAB brings two very basic `CurrencyProviders`.
Both support only one currency, but you can still use them for calculations.
The "locale base currency provider" selects the currency based on the configured locale, so in Germany the currency will be "EUR", in Denmark it will be "DKK".
The "fixes currency provider" allows you to define your own currency, you just have to select a three-letter code that identifies it.

There are plans to enhance that with support for multiple currency and automatic update of exchange rates.

## Main UI Enhancements
```